lib/ansible/plugins/strategy/linear.py

```diff
@@ -407,9 +407,12 @@ def run(self, iterator, play_context):
                         failed_hosts.append(res._host.name)
                     elif res.is_unreachable():
                         unreachable_hosts.append(res._host.name)
+                    elif any_errors_fatal and res.is_failed() and not iterator.is_failed(res._host):
+                        failed_hosts.append(res._host.name)
 
                 # if any_errors_fatal and we had an error, mark all hosts as failed
-                if any_errors_fatal and (len(failed_hosts) > 0 or len(unreachable_hosts) > 0):
+                # check TQM failed hosts in case a rescue/always was used after a failure in a block as failed_hosts may be empty
+                if any_errors_fatal and (len(failed_hosts) > 0 or len(unreachable_hosts) > 0 or len(self._tqm._failed_hosts) > 0):
```

Comment:
the new code will mark all hosts in the first play as failed, hence the 2nd play wont have any hosts to process.

I believe that is not exactly the behaviour we want, its probably closer to 'end play'  but leave the hosts that didn't fail available for the 2nd play. 

That said, having a play completely fail will already exit the playbook so that is probably fine, but the test seems to expect differently by adding a play that will never execute and trying to test a mid play failure preventing the last task from running.